src/peer-discovery/tests/index.js

```diff
@@ -1,17 +1,31 @@
 /* eslint-env mocha */
 'use strict'
 
+const chai = require('chai')
+const expect = chai.expect
+chai.use(require('dirty-chai'))
+
+const multiaddr = require('multiaddr')
+const PeerId = require('peer-id')
+
+const delay = require('delay')
+const pDefer = require('p-defer')
+
 module.exports = (common) => {
   describe('interface-peer-discovery', () => {
     let discovery
 
-    before(async () => {
+    beforeEach(async () => {
       discovery = await common.setup()
```

Comment:
If `common.setup()` needs to be run before every test, we need to tear it down after every test. The compliance test is causing this to leak intervals.